pkgs/development/go-modules/generic/default.nix

```
@@ -55,11 +58,24 @@
 with builtins;
 
 assert goPackagePath != "" -> throw "`goPackagePath` is not needed with `buildGoModule`";
+assert (vendorSha256 == "_unset" && vendorHash == "_unset") -> throw "either `vendorHash` or `vendorHash` is required";
+assert vendorSha256 != null && vendorSha256 != "_unset" && vendorHash != null && vendorHash != "_unset"
```

> I think this still allows for both to be set and null, if I'm not mistaken.

Sorry, I missed this the first time. As Sandro says don't we only want to allow one to be set? e.g.

```suggestion
assert (vendorSha256 != "_unset" && vendorHash != "_unset")
```